pkg/generic/thriftidl_provider.go

```diff
@@ -305,19 +327,22 @@ func NewThriftContentWithAbsIncludePathProviderWithDynamicGo(mainIDLPath string,
 	if !ok {
 		return nil, fmt.Errorf("miss main IDL content for main IDL path: %s", mainIDLPath)
 	}
-	svc, err := newServiceDescriptorFromContent(mainIDLPath, mainIDLContent, includes, true)
+	tOpts := &thriftIDLProviderOptions{}
+	tOpts.apply(opts)
+	parseMode := getParseMode(tOpts)
+	svc, err := newServiceDescriptorFromContent(mainIDLPath, mainIDLContent, includes, true, parseMode)
 	if err != nil {
 		return nil, err
 	}
 
-	p.newDynamicGoDsc(svc, mainIDLPath, mainIDLContent, includes)
+	p.newDynamicGoDsc(svc, mainIDLPath, mainIDLContent, includes, parseMode)
 
 	p.svcs <- svc
 	return p, nil
 }
 
 // UpdateIDL update idl by given args
-func (p *ThriftContentWithAbsIncludePathProvider) UpdateIDL(mainIDLPath string, includes map[string]string) error {
+func (p *ThriftContentWithAbsIncludePathProvider) UpdateIDL(mainIDLPath string, includes map[string]string, opts ...ThriftIDLProviderOption) error {
```

Comment:
I think no need to add opts to UpdateIDL interface, because it may be missing by users.